### PR TITLE
XW-669 | Create separate validator for Mercury Curated Content Editor

### PR DESCRIPTION
**Pull request by vforge**
see https://github.com/Wikia/app/pull/8975

**HAS TO BE RELEASED AFTER APP**

### CODE REVIEW

File: front/scripts/main/components/CuratedContentEditorComponent.js [hunk 1]
```diff
@@ -98,31 +99,4 @@ App.CuratedContentEditorComponent = Em.Component.extend(
...
-
-		/**
-		 * @param {string} type
-		 * @param {string} reason
-		 * @returns {void}
-		 */
-		processValidationError(type, reason) {
-			if (type === 'featured') {
-				this.addAlert({
-					message: i18n.t('app.curated-content-editor-error-inside-featured-content'),
```
Comment (rogatty) [inline front/scripts/main/components/CuratedContentEditorComponent.js]: Should we remove these strings from translations?

File: front/scripts/main/components/CuratedContentEditorComponent.js [hunk 2]
```diff
@@ -98,31 +99,4 @@ App.CuratedContentEditorComponent = Em.Component.extend(
...
-				// if other items occur that means user somehow bypassed validation of one or more items earlier
-				this.addAlert({
-					message: i18n.t('app.curated-content-editor-error-inside-items-message'),
-					type: 'alert'
-				});
-			}
-		}
-	}
-);
+	});
```
Comment (rogatty) [inline front/scripts/main/components/CuratedContentEditorComponent.js]: It's
```
(
    {
```
at the top so how about
```
    }
);
```
here?

File: front/scripts/main/components/CuratedContentEditorItemFormComponent.js [hunk 3]
```diff
@@ -428,11 +425,11 @@ App.CuratedContentEditorItemFormComponent = Em.Component.extend(
 		},
 
 		/**
-		 * @param {string} reason
+		 * @param {string} errorMessega
```
Comment (rogatty) [inline front/scripts/main/components/CuratedContentEditorItemFormComponent.js]: Typo

File: front/scripts/main/components/CuratedContentEditorItemFormComponent.js [hunk 4]
```diff
@@ -428,11 +425,11 @@ App.CuratedContentEditorItemFormComponent = Em.Component.extend(
...
 		 * @returns {void}
 		 */
-		processValidationError(reason) {
-			switch (reason) {
+		processValidationError(errorMessega) {
```
Comment (rogatty) [inline front/scripts/main/components/CuratedContentEditorItemFormComponent.js]: Typo

File: front/scripts/main/components/CuratedContentEditorComponent.js (hunk 1 above)
Comment (vforge) [inline front/scripts/main/components/CuratedContentEditorComponent.js]: Done in https://github.com/Wikia/mercury/commit/f805598bdf41c12f53899f28af37eccbb2d83dfd

File: front/scripts/main/components/CuratedContentEditorComponent.js [hunk 5]
```diff
@@ -98,31 +112,5 @@ App.CuratedContentEditorComponent = Em.Component.extend(
...
-		 */
-		processValidationError(type, reason) {
-			if (type === 'featured') {
-				this.addAlert({
-					message: i18n.t('app.curated-content-editor-error-inside-featured-content'),
-					type: 'alert'
-				});
-			} else if (reason === 'itemsMissing') {
-				this.addAlert({
-					message: i18n.t('app.curated-content-editor-missing-items-error'),
```
Comment (rogatty) [inline front/scripts/main/components/CuratedContentEditorComponent.js]: This one should be removed too, right?

File: front/scripts/main/components/CuratedContentEditorItemFormComponent.js [hunk 6]
```diff
@@ -396,18 +393,18 @@ App.CuratedContentEditorItemFormComponent = Em.Component.extend(
...
 			this.set('isLoading', true);
 
-			App.CuratedContentEditorItemModel.validateServerData(item, dataToValidate)
+			App.CuratedContentEditorItemModel.validateServerData(item, methodName)
 				.then((data) => {
 					if (data.status) {
 						this.sendAction('done', this.get('model'));
-					} if (data.error) {
-						data.error.forEach((error) => this.processValidationError(error.reason));
+					} else if (data.errors) {
```
Comment (rogatty) [inline front/scripts/main/components/CuratedContentEditorItemFormComponent.js]: `Em.isArray(data.errors)` would be safer.

File: front/scripts/main/components/CuratedContentEditorSectionComponent.js [hunk 7]
```diff
@@ -83,8 +82,8 @@ App.CuratedContentEditorSectionComponent = Em.Component.extend(
 						// Slice creates native JS array with only items (without meta properties).
 						this.set('model.items', sortableItems.slice(0, sortableItems.length));
 						this.sendAction('done', this.get('model'));
-					} else if (data.error) {
-						data.error.forEach((error) => this.processValidationError(error.reason));
+					} else if (data.errors) {
```
Comment (rogatty) [inline front/scripts/main/components/CuratedContentEditorSectionComponent.js]: Same as https://github.com/Wikia/mercury/pull/1610/files#r44129456

File: front/scripts/main/components/CuratedContentEditorSectionComponent.js [hunk 8]
```diff
@@ -105,11 +104,11 @@ App.CuratedContentEditorSectionComponent = Em.Component.extend(
 		},
 
 		/**
-		 * @param {string} reason
+		 * @param {string} errorMessega
 		 * @returns {void}
 		 */
-		processValidationError(reason) {
-			if (reason === 'itemsMissing') {
+		processValidationError(errorMessega) {
```
Comment (rogatty) [inline front/scripts/main/components/CuratedContentEditorSectionComponent.js]: Typo

File: front/scripts/main/components/CuratedContentEditorComponent.js (hunk 5 above)
Comment (bognix) [inline front/scripts/main/components/CuratedContentEditorComponent.js]: done

**Comment by tomasznapieralski** [comment]
lgtm

File: front/scripts/main/components/CuratedContentEditorComponent.js [hunk 9]
```diff
@@ -5,6 +5,19 @@ App.CuratedContentEditorComponent = Em.Component.extend(
...
+		/**
+		 * When user enters curated content editor we want to clear all notifications that might be still there
+		 * after previous edit. For example:
+		 * 1. user enters editor and taps publish - alert about successful page appears
+		 * 2. user gets redirected to main page and taps edit main page link
+		 * - alerts need to be reset because some of them might have not timed out
+		 *
+		 * @returns {void}
+		 */
+		didInsertElement() {
```
Comment (hakubo) [inline front/scripts/main/components/CuratedContentEditorComponent.js]: Call this. Super 
Or use ember.on

Comment (bognix) [inline front/scripts/main/components/CuratedContentEditorComponent.js]: I added this._super() call but interesting thing is that we are binding to this hook 24 times in our code but only twice we are calling this._super() (twice including this change)

The question is do we need it and if yes, should we fix it in this 23 places?

Comment (rogatty) [inline front/scripts/main/components/CuratedContentEditorComponent.js]: Yup, calling `this._super()` in `didInsertElement` is pointless. It makes sense in mixins though.

Anyway I support using `Em.on('didInsertElement', ...)` so we don't override anything from Ember.

**Pull request merged**